src/main/java/edu/wpi/grip/ui/PaletteView.java

```diff
@@ -37,9 +49,46 @@ public PaletteView(EventBus eventBus) {
             throw new RuntimeException(e);
         }
 
+
+        try {
+            setupClearButtonField(operationSearch);
+        } catch (Exception e) {
+            throw new IllegalStateException("Problem calling the setupClearButtonField method on TextFields", e);
+        }
+
+        final InvalidationListener filterOperations = observable ->
+                this.operations.getChildren().forEach(node -> {
+                    final Operation operation = ((OperationView) node).getOperation();
+                    node.setVisible(searchAlgorithmMatches(this.operationSearch.getText(), operation.getName()));
+                });
+
+        // Update the visibility of each node when either the list of nodes or the filter text changes
+        this.operations.getChildren().addListener(filterOperations);
+        this.operationSearch.textProperty().addListener(filterOperations);
+
         this.eventBus.register(this);
     }
 
+
+    // XXX: This is the only way to do this unfortunately.
+    // This is where this came from: https://bitbucket.org/controlsfx/controlsfx/issues/330/making-textfieldssetupclearbuttonfield
+    private void setupClearButtonField(CustomTextField customTextField) throws Exception {
+        Method m = TextFields.class.getDeclaredMethod("setupClearButtonField", TextField.class, ObjectProperty.class);
```

Comment:
No, this is pretty common practice to put a clear `x` button in the right side of an input box.